### PR TITLE
Feat/auth

### PR DESCRIPTION
**Pull request by Debarshi95**
PR for authentication.
This PR adds pages SignIn, Signup, Wishlist, Cart and various other components.
Handles authentication with backend apis. 
Uses JWT to authenticate user.

Deploy Preview: https://62409063b90e5344eea25fe6--debarshib-flixycart.netlify.app/

### CODE REVIEW

File: src/components/common/PrivateRoute/PrivateRoute.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+const PrivateRoute = ({ children, redirectTo }) => {
+  const { user } = useAuth();
+
+  if (!user) {
+    return <Navigate to={redirectTo} replace />;
+  }
+
+  return children;
+};
```
Comment (m26unkwn) [inline src/components/common/PrivateRoute/PrivateRoute.js:13]: You could have done this `user ? children: <Navigate to={} replace />`

anyhow above code is readable

**Review by m26unkwn — APPROVED** [review]
Code is clean and readable.

Few things that I like
1. Using code splitting 
2. Using private route

File: src/pages/auth/Signup/Signup.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import { Formik, Form } from 'formik';
+import { Button, Input, Typography } from '../../../components';
+import { signUp } from '../../../services/flixycartApi';
+import { validateRegister } from '../../../utils/validations';
+import { useAuth } from '../../../providers/AuthProvider/AuthProvider';
+import { setItem } from '../../../utils/helperFuncs';
+import './Signup.scss';
+
+const Signup = () => {
+  const { setUser } = useAuth();
```
Comment (dev-ashishsethi) [inline src/pages/auth/Signup/Signup.js:12]: good that you are using context for this

File: src/pages/auth/Signin/Signin.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                    type="submit"
+                    variant="contained"
+                    className="Signin__button text-bold"
+                    disabled={
+                      isSubmitting ||
+                      !isValid ||
+                      Boolean(values.email === '' || values.password === '')
+                    }
+                  >
+                    {isSubmitting ? 'Submitting...' : 'Login'}
```
Comment (dev-ashishsethi) [inline src/pages/auth/Signin/Signin.js:94]: you didn't do validation of inputs

**Review by dev-ashishsethi — APPROVED** [review]
do check comments

File: src/pages/auth/Signin/Signin.js (hunk 3 above)
Comment (Debarshi95) [reply src/pages/auth/Signin/Signin.js:94]: Actually I did. It's done internally through formik. The validationSchema in Formik is actually validating the inputs

**Pull request merged**